File: src/1184-Distance-Between-Bus-Stops/1184.cpp
iclass Solution 
{
public:
    int distanceBetweenBusStops(vector<int>& d, int l, int r) 
    {
        if (l > r) swap(l, r);
        int k = 0, v = 0;
        for (int i = 0; i < d.size(); i++)
        {
            if (i < l or i >= r) k += d[i];
            else v += d[i];
        }
        return min(k, v);
    }
};

File: src/1184-Distance-Between-Bus-Stops/1184.go
func distanceBetweenBusStops(d []int, l int, r int) int {
    if l > r {
        l, r = r, l
    }
    k, v := 0, 0
    for i, t := range d {
        if i < l || i >= r {
            k += t
        } else {
            v += t
        }
    }
    
    return min(k, v)
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}